src/coverlet.template/ModuleTrackerTemplate.cs

```diff
@@ -52,6 +52,30 @@ public static void RecordHit(int hitLocationIndex)
             Interlocked.Increment(ref HitsArray[hitLocationIndex]);
         }
 
+        public static void RecordHit(int hitLocationIndex)
+        {
+            Interlocked.Increment(ref HitsArray[hitLocationIndex]);
+        }
+
+        public static void RecordSingleHitInCoreLibrary(int hitLocationIndex)
+        {
+            // Make sure to avoid recording if this is a call to RecordHit within the AppDomain setup code in an
+            // instrumented build of System.Private.CoreLib.
+            if (HitsArray is null)
+                return;
+
+            ref int location = ref HitsArray[hitLocationIndex];
+            if (location == 0)
+                location = 1;
+        }
+
+        public static void RecordSingleHit(int hitLocationIndex)
+        {
+            ref int location = ref HitsArray[hitLocationIndex];
+            if (location == 0)
+                location = 1;
```

Comment:
📝 The main value here is avoiding this write invalidating cache lines. OpenCover uses a configurable threshold value. I could rewrite it to use a configurable threshold, though most uses I've seen in practice either omit this value or set it to 1.

Reply:
Why don't pass `_singleHit` as parameter or use static var and have less methods?Inlining?

Reply:
It's always a possibility to come back and do this if it proves to be equally fast. Since the expanded form is at least that fast and not difficult to work with, it seemed like the straightforward initial approach.